src/message_filters/__init__.py

```diff
@@ -275,7 +291,7 @@ def add(self, msg, my_queue, my_queue_index=None):
                 return
             topic_stamps = sorted(topic_stamps, key=lambda x: x[1])
             stamps.append(topic_stamps)
-        for vv in itertools.product(*[zip(*s)[0] for s in stamps]):
+        for vv in itertools.product(*[list(zip(*s))[0] for s in stamps]):
```

Comment:
Why this change? The output of zip is an iterable and converting it to a list doesn't seem to add functionality.

Reply:
There is the following error which says 'zip' object is not subscriptable for test `test_approxsync.py`

```
12:             self.assertEqual(self.collector, [])
12:             for msg in random.sample(seq1, N):
12: >               m1.signalMessage(msg)
12:
12: test/test_approxsync.py:101:
12: _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
12: ../../build/message_filters/install/message_filters/lib/python3.6/site-packages/message_filters/__init__.py:63: in signalMessage
12:     cb(*(msg + args))
12: ../../build/message_filters/install/message_filters/lib/python3.6/site-packages/message_filters/__init__.py:295: in add
12:     for vv in itertools.product(*[zip(*s)[0] for s in stamps]):
12: _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
12:
12: .0 = <list_iterator object at 0x7f0dcde86908>
12:
12: >   for vv in itertools.product(*[zip(*s)[0] for s in stamps]):
12: E   TypeError: 'zip' object is not subscriptable
12:
12: ../../build/message_filters/install/message_filters/lib/python3.6/site-packages/message_filters/__init__.py:295: TypeError
12: =========================== 1 failed in 0.08 seconds ===========================
12: -- run_test.py: return code 1
12: -- run_test.py: verify result file '/home/ethan/Myspace/rosdev/2upstream/build/message_filters/build/message_filters/test_results/message_filters/test_approxsync.py.xunit.xml'
12/13 Test #12: test_approxsync.py .............................***Failed    0.94 sec

```

Reply:
Ahh apparently that's a python2 python3 change: https://stackoverflow.com/a/27431433/604099